sources/web/datalab/server.ts

```diff
@@ -267,19 +272,31 @@ function stopVmHandler(request: http.ServerRequest, response: http.ServerRespons
 }
 
 function socketHandler(request: http.ServerRequest, socket: net.Socket, head: Buffer) {
+  request.url = trimBasePath(request.url);
   // Avoid proxying websocket requests on this path, as it's handled locally rather than by Jupyter.
   if (request.url != httpOverWebSocketPath) {
     jupyter.handleSocket(request, socket, head);
   }
 }
 
+function trimBasePath(path: string): string {
+  let pathPrefix = appSettings.datalabBasePath;
+  if (path.indexOf(pathPrefix) == 0) {
+    let newPath = "/" + path.substring(pathPrefix.length);
```

Comment:
Or you could skip the "/"+ part and instead subtract 1 from the length you are passing to substring, since you know the pathPrefix ends with a slash. Your choice.